Add documentation for peer service mapping (#3428)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/java/automatic/agent-config.md

`cats-service` and requests to `dogs-abcdef123.serverlessapis.com` will have an
attribute of `dogs-api`.

> Since javaagent version `1.31.0`, it is possible to provide a port and a path to

Reviewer: ```suggestion
Since Java agent version `1.31.0`, it is possible to provide a port and a path to
```

define a `peer.service`.

> For example

Reviewer: ```suggestion
For example, if set to the following:
```

    1.2.3.4:443=cats-service,dogs-abcdef123.serverlessapis.com:80/api=dogs-api

> Then, requests to `1.2.3.4` will have no override for `peer.service` attribute
> while `1.2.3.4:443` will have have `peer.service` of `cats-service` and requests
> to `dogs-abcdef123.serverlessapis.com:80/api/v1` will have an attribute of
> `dogs-api`.

Reviewer: ```suggestion
Then, requests to `1.2.3.4` will have no override for `peer.service` attribute,
while `1.2.3.4:443` will have have `peer.service` of `cats-service` and requests
to `dogs-abcdef123.serverlessapis.com:80/api/v1` will have an attribute of
`dogs-api`.
```
